### PR TITLE
cmd/bosun: Add over, shift, and merge funcs

### PR DESCRIPTION
**Pull request by kylebrandt**
This adds the ability to graph multiple seriesSets in expression graphs, and also graph things like week-over-week.

### CODE REVIEW

File: cmd/bosun/expr/funcs.go [hunk 1]
```diff
@@ -371,6 +396,43 @@ func Filter(e *State, T miniprofiler.Timer, series *Results, number *Results) (*
 	return series, nil
 }
 
+func Merge(e *State, T miniprofiler.Timer, series ...*Results) (*Results, error) {
+	if len(series) < 1 {
```
Comment (captncraig) [inline cmd/bosun/expr/funcs.go]: I'd prefer `== 0` and `== 1` for these two short circuit clauses.

File: cmd/bosun/expr/funcs.go [hunk 2]
```diff
@@ -732,6 +796,91 @@ func Band(e *State, T miniprofiler.Timer, query, duration, period string, num fl
 	return
 }
 
+func ShiftBand(e *State, T miniprofiler.Timer, query, duration, period string, num float64) (r *Results, err error) {
```
Comment (captncraig) [inline cmd/bosun/expr/funcs.go]: does this function need to be exported?

Comment (captncraig) [inline cmd/bosun/expr/funcs.go]: It looks like it should be, but it is not documented.

Comment (kylebrandt) [inline cmd/bosun/expr/funcs.go]: By convention, in funcs.go, all the captial functions work with Sets, and the lowercase ones work within results within the set. So I am following that.

**Pull request merged**